test/units/modules/test_iptables.py

```diff
@@ -1,1174 +1,1402 @@
-            '-L',
-            'INPUT',
-        ])
-        self.assertEqual(run_command.call_args_list[1][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-P',
-            'INPUT',
-            'ACCEPT',
-        ])
-
-    def test_policy_table_no_change(self):
-        """Test don't change policy of a chain if the policy is right"""
-        set_module_args({
-            'policy': 'ACCEPT',
-            'chain': 'INPUT',
-        })
-        commands_results = [
-            (0, 'Chain INPUT (policy ACCEPT)\n', ''),
-            (0, '', '')
-        ]
-
-        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
-            run_command.side_effect = commands_results
-            with self.assertRaises(AnsibleExitJson) as result:
-                iptables.main()
-                self.assertFalse(result.exception.args[0]['changed'])
-
-        self.assertEqual(run_command.call_count, 1)
-        self.assertEqual(run_command.call_args_list[0][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-L',
-            'INPUT',
-        ])
-
-    def test_policy_table_changed_false(self):
-        """Test flush without parameters and change == false"""
-        set_module_args({
-            'policy': 'ACCEPT',
-            'chain': 'INPUT',
-            '_ansible_check_mode': True,
-        })
-        commands_results = [
-            (0, 'Chain INPUT (policy DROP)\n', ''),
-        ]
-
-        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
-            run_command.side_effect = commands_results
-            with self.assertRaises(AnsibleExitJson) as result:
-                iptables.main()
-                self.assertTrue(result.exception.args[0]['changed'])
-
-        self.assertEqual(run_command.call_count, 1)
-        self.assertEqual(run_command.call_args_list[0][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-L',
-            'INPUT',
-        ])
+    """Return iptables binary path."""
+    return IPTABLES_CMD
+
+
+@pytest.fixture
+def mock_basic_commands(mocker):
+    """Mock basic commands like get_bin_path and get_iptables_version."""
+    mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.get_bin_path",
+        return_value=get_bin_path(),
+    )
+    mocker.patch("ansible.modules.iptables.get_iptables_version", return_value=IPTABLES_VERSION)
+
+
+def test_without_required_parameters(mocker):
+    """Test module without any parameters."""
+    mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.fail_json",
+        side_effect=fail_json,
+    )
+    set_module_args({})
+    with pytest.raises(AnsibleFailJson) as exc:
+        iptables.main()
+
+    assert exc.value.args[0]["failed"]
+    assert "Failed to find required executable" in exc.value.args[0]["msg"]
+
+
+def test_flush_table_without_chain(mocker, mock_basic_commands):
```

Comment:
So whenever it's not necessary to access the fixture return/yield values, those are cleaner to list through the decorator rather than inject as arguments.
```suggestion
@pytest.mark.usefixtures('_mock_basic_commands')
def test_flush_table_without_chain(mocker):
```

I learned this from https://pypi.org/p/flake8-pytest-style

Reply:
TIL, thanks.